src/test/scala/mesosphere/mesos/ConstraintsTest.scala

```diff
@@ -645,6 +645,30 @@ class ConstraintsTest extends UnitTest {
     }
   }
 
+  "IS operator" should {
+    "require that a value match exactly" in {
+      makeOffer("righthost.com") should meetConstraint(hostnameField, IS, "righthost.com")
+      makeOffer("wronghost.com") should meetConstraint(hostnameField, IS, "wronghost.com")
+    }
+  }
+
+  "IN operator" should {
+    "require that the offer value be in the comma delimited list of values" in {
+      makeOffer("host1") should meetConstraint(hostnameField, IN, "host1,host2")
```

Comment:
I am worried that values might have commas in them. In this case, we would need to support escaping and/or strings inside value strings of constraint definitions. Although I would vote for supports of arrays of strings as constraint values. In this case the constraint in an app definition could look like `["some-field", "IN", ["value1", "value2"]`. What do you think?

Reply:
I like what Ivan is proposing! :)

Reply:
This would break API compatibility

Reply:
Please forgive me my ignorance, but I do not see how it would break the user-facing API compatibility. Isn't it about addition/extension only? Are you referring to the case when Java/Python/etc clients would fail if their receive an array value, and not a string one?

Reply:
It would change the type, specified here:

https://github.com/mesosphere/marathon/blob/v1.6.0-pre/docs/docs/rest-api/public/api/v2/types/constraint.raml#L20-L29

The specification is that constraints are a 2-3 item array of strings. Putting a non-string in there breaks the API.

Reply:
I was thinking of supporting strings _and_ arrays: arrays for `IN` and strings for everything else.

Reply:
I understood this :)

Reply:
@ichernetsky did you mean to write this earlier?

```
["some-field", "IN", "value1", "value2"]
```

(sans the additional `[`)?

@jdef mentioned this as a possibility and it made me wonder if perhaps this was what you were trying to suggest :)

Reply:
No, I was suggesting `["field", "IN", ["value1", "value2"]]`.